docs/docs/develop/custom-apis/index.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Custom API"
+description: Create your own API to pull data out in flexible manner 
+sidebar_label: "Custom API"
+sidebar_position: 20
+---
+
+Rill allows you to create custom APIs to pull data out in a flexible manner. You can write custom SQL queries and expose them as an API endpoint.
+
+## Create a custom API
+
+To create a custom API, create a new yaml file under `apis` directory in your Rill project. Currently, 
+we support two types of custom APIs:
+
+1. **SQL API**: You can write a SQL query and expose it as an API endpoint. This is useful when you want to directly 
+    write queries against a [model](../sql-models.md) that you have created. It should have the following structure:
```

Comment:
technically a user can also query source right ? Are we not documenting that ?

Reply:
yeah I was trying to avoid that since for druid there are no sources and in the metrics view we put datasource name under `model` so was not sure. Now I think about it, it should be ok to add that. What do you think ?

Reply:
added it for now